### PR TITLE
[DOCS] Changes wording to move away from data frame terminology in the ES repo

### PR DESCRIPTION
**Pull request by szabosteve**
Related to https://github.com/elastic/stack-docs/issues/518

This PR changes the wording of some documentation pieces to move away from the data frame terminology.

Changed files:

```
docs/java-rest/high-level/transform/stop_transform.asciidoc 
docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc 
docs/reference/ml/df-analytics/apis/evaluateresources.asciidoc 
docs/reference/settings/ml-settings.asciidoc 
docs/reference/transform/examples.asciidoc 
docs/reference/transform/limitations.asciidoc
```

Other occurrences of the `data frame` or `data frames` expressions were necessary to leave in the text.

List of occurrences:

```
find docs -name '*.asciidoc' | xargs egrep 'data frame'
docs/reference/settings/ml-settings.asciidoc:`20`. In this context, jobs include both anomaly detector jobs and data frame
docs/reference/transform/overview.asciidoc:image::images/ml-dataframepivot.jpg["Example of a data frame pivot in {kib}"]

find docs -name '*.asciidoc' | xargs egrep 'dataframe'
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-get-data-frame-transform"]
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-get-data-frame-transform-stats"]
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-put-data-frame-transform"]
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-update-data-frame-transform"]
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-delete-data-frame-transform"]
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-preview-data-frame-transform"]
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-start-data-frame-transform"]
docs/java-rest/redirects.asciidoc:[role="exclude",id="java-rest-high-dataframe-stop-data-frame-transform"]
docs/java-rest/high-level/ml/evaluate-data-frame.asciidoc:Evaluates the {ml} algorithm that ran on a {dataframe}.
docs/java-rest/high-level/ml/estimate-memory-usage.asciidoc:<1> Constructing a new request containing a {dataframe-analytics-config} for which memory usage estimation should be performed
docs/reference/ml/df-analytics/apis/estimate-memory-usage-dfanalytics.asciidoc:Estimates memory usage for the given {dataframe-analytics-config}.
docs/reference/ml/df-analytics/apis/estimate-memory-usage-dfanalytics.asciidoc:This API estimates memory usage for the given {dataframe-analytics-config} before the {dfanalytics-job} is even created.
docs/reference/ml/df-analytics/apis/evaluateresources.asciidoc:the probability that each {dataframe} row belongs to a certain class. For 
docs/reference/ml/df-analytics/apis/dfanalyticsresources.asciidoc:  (Required, string) Defines which field of the {dataframe} is to be predicted. 
docs/reference/transform/checkpoints.asciidoc:. Updates the destination index (the {dataframe}) with the changed entities.
docs/reference/transform/apis/transformresource.asciidoc:{stack-ov}/ecommerce-dataframes.html[Transforming your data with {dataframes}].
docs/reference/transform/apis/transformresource.asciidoc:{stack-ov}/dataframe-limitations.html[{dataframe-cap} limitations].
docs/reference/transform/apis/put-transform.asciidoc:structure (known as a {dataframe}). The ID for each document in the
docs/reference/transform/apis/put-transform.asciidoc:{dataframe} is generated from a hash of the entity, so there is a unique row
docs/reference/transform/limitations.asciidoc:the Elastic {dataframe} feature:
docs/reference/transform/limitations.asciidoc:==== {dataframe-cap} UI will not work during a rolling upgrade from 7.2
docs/reference/transform/limitations.asciidoc:created in 7.2, the {dataframe} UI will not work. Please wait until all nodes 
docs/reference/transform/limitations.asciidoc:have been upgraded to the newer version before using the {dataframe} UI.
docs/reference/transform/limitations.asciidoc:==== {dataframe-cap} data type limitation
docs/reference/transform/limitations.asciidoc:{dataframes-cap} do not (yet) support fields containing arrays – in the UI or 
docs/reference/transform/limitations.asciidoc:(deleted, updated, added) while the batch {dataframe} is in progress, then the 
docs/reference/transform/limitations.asciidoc:==== {cdataframe-cap} consistency does not account for deleted or updated documents
docs/reference/transform/limitations.asciidoc:will not be removed from the {dataframe} destination index.
docs/reference/transform/limitations.asciidoc:updated when viewing the {dataframe} destination index.
docs/reference/transform/limitations.asciidoc:==== Deleting a {transform} does not delete the {dataframe} destination index or {kib} index pattern
docs/reference/transform/limitations.asciidoc:neither the {dataframe} destination index nor the {kib} index pattern, should 
docs/reference/transform/limitations.asciidoc:{transform} checkpoint to complete. For {cdataframes}, the number of 
docs/reference/transform/limitations.asciidoc:==== {cdataframe-cap} scheduling limitations
docs/reference/transform/limitations.asciidoc:A {cdataframe} periodically checks for changes to source data. The functionality 
docs/reference/transform/limitations.asciidoc:==== {cdataframes-cap} may give incorrect results if documents are not yet available to search
docs/reference/transform/overview.asciidoc:A _{dataframe}_ is a two-dimensional tabular data structure. In the context of
docs/reference/transform/overview.asciidoc:example, you can use {dataframes} to _pivot_ your data into a new entity-centric
docs/reference/transform/overview.asciidoc:individual document, for example a single item purchase. {dataframes-cap} enable
docs/reference/transform/overview.asciidoc:You can create {dataframes} by using {transforms}.
docs/reference/transform/overview.asciidoc:Pivoting results in a summary of your data, which is the {dataframe}.
docs/reference/transform/overview.asciidoc:ordered quantity. The result is a {dataframe} that shows the number of sold
docs/reference/transform/overview.asciidoc:image::images/ml-dataframepivot.jpg["Example of a data frame pivot in {kib}"]
docs/reference/transform/overview.asciidoc:creates a new index that is dedicated to the {dataframe}.
docs/reference/transform/usage.asciidoc:joined-up picture. This new index is sometimes referred to as a _{dataframe}_.
docs/reference/transform/usage.asciidoc:all aggregation results, you need to create the complete {dataframe}. If you
docs/reference/transform/ecommerce-tutorial.asciidoc:If you want to use more complex queries, you can create your {dataframe} from a
docs/reference/transform/ecommerce-tutorial.asciidoc:image::images/dataframe-transforms.jpg["Managing {transforms} in {kib}"]
docs/reference/transform/examples.asciidoc:<1> This is the destination index for the {dataframe}. It is ignored by 
docs/reference/transform/examples.asciidoc:<2> Two `group_by` fields have been selected. This means the {dataframe} will 
docs/reference/transform/examples.asciidoc:dataset both these fields are unique. By including both in the {dataframe} it 
docs/reference/transform/examples.asciidoc:{dataframe} in advance, populated with some sample values. For example:
docs/reference/transform/examples.asciidoc:This {dataframe} makes it easier to answer questions such as:
docs/reference/transform/examples.asciidoc:{dataframes} allow us to persist this data as a customer centric index. This 
docs/reference/transform/examples.asciidoc:<2> This is the destination index for the {dataframe}. It is ignored by 
docs/reference/transform/examples.asciidoc:This {dataframe} makes it easier to answer questions such as:
docs/reference/transform/examples.asciidoc:For batch {dataframes} this occurs once.
docs/reference/transform/examples.asciidoc:<2> This is the destination index for the {dataframe}. It is ignored by 
docs/reference/transform/examples.asciidoc:This {dataframe} makes it easier to answer questions such as:
```

### CODE REVIEW

File: docs/reference/transform/examples.asciidoc [hunk 1]
```diff
@@ -275,9 +275,9 @@ POST _data_frame/transforms/_preview
 ----------------------------------
 // TEST[skip:setup kibana sample data]
 
-<1> This range query limits the {transform} to documents that are within the last 
-30 days at the point in time the {transform} checkpoint is processed. 
-For batch {dataframes} this occurs once.
+<1> This range query limits the {transform} to documents that are within the 
+last 30 days at the point in time the {transform} checkpoint is processed. For 
+batch {dataframes} this occurs once.
```
Comment (lcawl) [inline docs/reference/transform/examples.asciidoc]:
```suggestion
batch {transforms} this occurs once.
```

File: docs/reference/transform/limitations.asciidoc [hunk 2]
```diff
@@ -8,30 +8,30 @@
...
 [[transform-ui-limitation]]
 ==== {dataframe-cap} UI will not work during a rolling upgrade from 7.2
 
 If your cluster contains mixed version nodes, for example during a rolling 
-upgrade from 7.2 to a newer version, and {transforms} have been 
-created in 7.2, the {dataframe} UI will not work. Please wait until all nodes 
-have been upgraded to the newer version before using the {dataframe} UI.
+upgrade from 7.2 to a newer version, and {transforms} have been created in 7.2, 
+the {dataframe} UI will not work. Please wait until all nodes have been upgraded 
+to the newer version before using the {dataframe} UI.
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]: I think at some point the "Dataframe UI" became the "Transforms UI".  Should we clarify that here?

File: docs/reference/transform/limitations.asciidoc [hunk 3]
```diff
@@ -87,48 +90,49 @@ do not yet support a search context, therefore if the source data is changed
...
-Depending on your use case, you may wish to recreate the {transform} 
-entirely after deletions. Alternatively, if your use case is tolerant to 
-historical archiving, you may wish to include a max ingest timestamp in your 
-aggregation. This will allow you to exclude results that have not been recently 
-updated when viewing the {dataframe} destination index.
+Depending on your use case, you may wish to recreate the {transform} entirely 
+after deletions. Alternatively, if your use case is tolerant to historical 
+archiving, you may wish to include a max ingest timestamp in your aggregation. 
+This will allow you to exclude results that have not been recently updated when 
+viewing the {dataframe} destination index.
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]:
```suggestion
viewing the {transform} destination index.
```

File: docs/reference/transform/limitations.asciidoc [hunk 4]
```diff
@@ -87,48 +90,49 @@ do not yet support a search context, therefore if the source data is changed
...
 
+
 [float]
 [[transform-consistency-limitations]]
 ==== {cdataframe-cap} consistency does not account for deleted or updated documents
 
-While the process for {transforms} allows the continual recalculation 
-of the {transform} as new data is being ingested, it does also have 
-some limitations.
+While the process for {transforms} allows the continual recalculation of the
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]: Should the title of this section be changed to {ctransform-cap}?

File: docs/reference/transform/limitations.asciidoc [hunk 5]
```diff
@@ -138,29 +142,28 @@ calculated based upon all activity within the cluster, not just activity from
...
 
-For a batch {transform}, the number of buckets requested is only ever 
-adjusted downwards. The lowering of value may result in a longer duration for the 
-{transform} checkpoint to complete. For {cdataframes}, the number of 
-buckets requested is reset back to its default at the start of every checkpoint 
-and it is possible for circuit breaker exceptions to occur repeatedly in the 
-{es} logs. 
+For a batch {transform}, the number of buckets requested is only ever adjusted 
+downwards. The lowering of value may result in a longer duration for the 
+{transform} checkpoint to complete. For {cdataframes}, the number of buckets
```
Comment (lcawl) [inline docs/reference/transform/limitations.asciidoc]: I don't think it's mandatory, but there's a newer attribute:
```suggestion
{transform} checkpoint to complete. For {ctransforms}, the number of buckets 
```

File: docs/reference/transform/limitations.asciidoc (hunk 2 above)
Comment (szabosteve) [reply docs/reference/transform/limitations.asciidoc]: Sure, thanks!

**Comment by szabosteve** [comment]
@lcawl Can you approve this PR if you find it okay? Then we could close the related issue.

**Review by lcawl — APPROVED** [review]
LGTM!

**Comment by szabosteve** [comment]
Backported to 7.x, 7.4, and 7.3.

**Pull request merged**